jenkins/jobs/image_building.pipeline

```diff
@@ -15,61 +14,88 @@ pipeline {
   }
   stages {
     stage('SCM') {
-      options {
-        timeout(time: 5, unit: 'MINUTES')
-      }
-      steps {
-        /* Checkout CI Repo */
-        checkout([$class: 'GitSCM',
-                 branches: [[name: ci_git_branch]],
-                 doGenerateSubmoduleConfigurations: false,
-                 extensions: [[$class: 'WipeWorkspace'],
-                 [$class: 'CleanCheckout'],
-                 [$class: 'CleanBeforeCheckout']],
-                 submoduleCfg: [],
-                 userRemoteConfigs: [[credentialsId: ci_git_credential_id,
-                 url: ci_git_url]]])
-        script {
-          CURRENT_START_TIME = System.currentTimeMillis()
-        }
-      }
-    }
-    stage('Build Ubuntu CI image') {
-      options {
-        timeout(time: 3, unit: 'HOURS')
-      }
-      environment {
-        IMAGE_OS = "ubuntu"
-      }
-      steps {
-        echo 'Building Ubuntu CI image'
-        withCredentials([
-          usernamePassword(credentialsId: 'xerces-est-metal3ci', usernameVariable: 'OPENSTACK_USERNAME_XERCES', passwordVariable: 'OPENSTACK_PASSWORD_XERCES'),
-          usernamePassword(credentialsId: 'metal3ci_city_cloud_openstack_credentials', usernameVariable: 'OPENSTACK_USERNAME_CLEURA', passwordVariable: 'OPENSTACK_PASSWORD_CLEURA'),
-          usernamePassword(credentialsId: 'infra-nordix-artifactory-api-key', usernameVariable: 'RT_USER', passwordVariable: 'RT_TOKEN')
-        ]) {
-          catchError([stageResult: 'FAILURE', message: "Failed to build Ubuntu CI image"]) {
-            sh "./jenkins/image_building/build-image.sh"
+      matrix {
+        agent { label "metal3ci-jenkins" }
+        options { ansiColor('xterm') }
+        axes {
+          axis {
+            name 'IMAGE_OS'
+            values 'ubuntu', 'centos'
           }
         }
-      }
-    }
-    stage('Building Centos CI image'){
-      options {
-        timeout(time: 3, unit: 'HOURS')
-      }
-      environment {
-        IMAGE_OS = "centos"
-      }
-      steps {
-        echo 'Building Centos CI image'
-        withCredentials([
-          usernamePassword(credentialsId: 'xerces-est-metal3ci', usernameVariable: 'OPENSTACK_USERNAME_XERCES', passwordVariable: 'OPENSTACK_PASSWORD_XERCES'),
-          usernamePassword(credentialsId: 'metal3ci_city_cloud_openstack_credentials', usernameVariable: 'OPENSTACK_USERNAME_CLEURA', passwordVariable: 'OPENSTACK_PASSWORD_CLEURA'),
-          usernamePassword(credentialsId: 'infra-nordix-artifactory-api-key', usernameVariable: 'RT_USER', passwordVariable: 'RT_TOKEN')
-        ]) {
-          catchError([stageResult: 'FAILURE', message: "Failed to build Centos CI image"]) {
-            sh "./jenkins/image_building/build-image.sh"
+        environment {
+          IMAGE_OS = "${IMAGE_OS}"
+        }
+        stages {
+          stage("Checkout CI Repo") {
```

Comment:
Can we merge stage("Checkout CI Repo") and stage("Build CI image"). Checking out does not necessarily need to have its own stage. We are having too many granular steps on the view.  

![image](https://github.com/user-attachments/assets/53eb45c5-d02b-47ff-9baa-31920593c2c2)

Reply:
I have no big issue, but I tend towards disagreeing with that. We've most likely always put the checkout as a step in all of our jobs, I'd prefer to keep it that way for consistency.